src/test/java/parser/JsonParserTest.java

```diff
@@ -0,0 +1,151 @@
+package parser;
+
+import com.github.javafaker.Faker;
+import com.google.gson.Gson;
+import org.apache.commons.lang3.StringUtils;
+import org.testng.Assert;
+import org.testng.annotations.*;
+import org.testng.asserts.SoftAssert;
+import shop.Cart;
+import shop.RealItem;
+import shop.VirtualItem;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Locale;
+
+import static org.testng.Assert.assertThrows;
+
+public class JsonParserTest {
+
+    public static final String RESOURCE_PATH = "src/main/resources/%s.json";
+    private JsonParser jsonParser;
+    private Gson gson;
+    private static String cartName;
+    private static String filePath;
+    private static Faker faker;
+    private static List<String> listOfFiles;
+
+    @BeforeClass
+    public void init() {
+        jsonParser = new JsonParser();
+        faker = new Faker();
+        gson = new Gson();
+        cartName = faker.name().firstName().toLowerCase(Locale.ROOT);
+        filePath = String.format(RESOURCE_PATH, cartName);
+        listOfFiles = new ArrayList<>();
+        listOfFiles.add(filePath);
+    }
+
+    @Test(description = "JSON Parser test - Write to existing file")
+    public void parserWritePositive() throws IOException {
+        Cart cart = new Cart(cartName);
+
+        VirtualItem laptop = new VirtualItem();
+        double laptopPrice = 567.0;
+        laptop.setPrice(laptopPrice);
+        cart.addVirtualItem(laptop);
+        double actualTotalPrice = cart.getTotalPrice();
+        double expectedTotalPrice = laptopPrice + laptopPrice * 0.2;
+
+        jsonParser.writeToFile(cart);
+        String actualData = readFromInputStream();
```

Comment:
Once you know actual result, you can get total price (actual price). Please, let me know, if you will have questions